### PR TITLE
Added Two-factor authentication API documentation

### PR DESCRIPTION
**Pull request by aleksandrychev**
Ticket: ENT-10838

### CODE REVIEW

**Comment by aleksandrychev** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/605/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/605/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-605/output/_site/

File: api/enterprise-api-ref/two-factor-authentication.markdown [hunk 1]
```diff
@@ -0,0 +1,239 @@
+---
+layout: default
+title: Two-factor authentication API
+published: true
+---
+The Two-factor authentication API enables users to add an extra layer of security to their accounts
+by requiring a TOTP (time-based one-time password) in addition to their primary credentials.
+
+Once two-factor authentication is enabled, requests to obtain an OAuth token, change a password, or
+disable two-factor authentication must include an additional header: Cf-2fa-Token: code.
```
Comment (craigcomstock) [inline api/enterprise-api-ref/two-factor-authentication.markdown:10]:
```suggestion
disable two-factor authentication must include an additional header: `Cf-2fa-Token: <code>`.
```

Maybe make it more clear that the head includes both the name: `Cf-2fa-Token` and a value?

File: api/enterprise-api-ref/two-factor-authentication.markdown [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+**Successful response example:**
+
+```
+HTTP 200 Ok
+{
+    "secret": "SYSZJJVQUIOG5ITA6HPH5PZWLZ3VUP52",
+    "2faUrl": "otpauth://totp/Mission%20Portal:admin?secret=SYSZJJVQUIOG5ITA6HPH5PZWLZ3VUP52&issuer=Mission%20Portal&algorithm=SHA1&digits=6&period=30",
+    "algorithm": "sha1",
+    "digits": 6,
```
Comment (craigcomstock) [inline api/enterprise-api-ref/two-factor-authentication.markdown:42]: can we include a description of the fields?

digits?
period is... days?
holder is username?

File: api/enterprise-api-ref/users-rbac.markdown [hunk 3]
```diff
@@ -74,6 +76,8 @@ API call allowed only for administrator.
     List of assigned RBAC roles.
 * **external**
     Is user from external source (LDAP/AD).
+* **two_factor_enabled**
+    Is a user has enabled two-factor authentication
```
Comment (craigcomstock) [inline api/enterprise-api-ref/users-rbac.markdown:80]:
```suggestion
    If a user has enabled two-factor authentication
```

File: api/enterprise-api-ref/users-rbac.markdown [hunk 4]
```diff
@@ -123,6 +128,8 @@ API call allowed only for administrator.
     Is user from external source (LDAP/AD).
 * **time_zone**
     Time zone
+* **two_factor_enabled**
+    Is a user has enabled two-factor authentication
```
Comment (craigcomstock) [inline api/enterprise-api-ref/users-rbac.markdown:132]:
```suggestion
    If a user has enabled two-factor authentication
```

**Review by craigcomstock — COMMENTED** [review]
looks good, a few little comments

File: api/enterprise-api-ref/two-factor-authentication.markdown (hunk 1 above)
Comment (nickanderson) [reply api/enterprise-api-ref/two-factor-authentication.markdown:10]: +1

File: api/enterprise-api-ref/two-factor-authentication.markdown [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+disable two-factor authentication must include an additional header: Cf-2fa-Token: code.
+Basic authentication also requires this header each time it is used.
+
+Currently, only TOTP (Time-based One-Time Password) two-factor authentication
+is supported, providing users with a time-sensitive code for enhanced security.
+
+## Start two-factor authentication configuration
+
+First, you need to request two-factor authentication secrets to complete the configuration.
+The secret token should be added to a third-party authentication app such as Google Authenticator or Authy.
```
Comment (nickanderson) [inline api/enterprise-api-ref/two-factor-authentication.markdown:19]: no need to add it but i had the urge to add totp-cli :D

File: api/enterprise-api-ref/two-factor-authentication.markdown (hunk 2 above)
Comment (nickanderson) [reply api/enterprise-api-ref/two-factor-authentication.markdown:42]: Good suggestion.

File: api/enterprise-api-ref/two-factor-authentication.markdown [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+|---------------------------|----------------------------------------|
+| 200 OK                    | 2FA successfully disabled              |
+| 401 Unauthorized          | Invalid two-factor authentication code |
+| 409 Conflict              | 2FA is not enabled for this user       |
+| 500 Internal server error | Internal server error                  |
+
+## Disable two-factor authentication for other users
+
+In case if a regular user loses access to the authentication application, specific users (admin role by default)
+with the `Disable 2FA for any user` RBAC role (alias 2fa.any-user.disable) will be able to
```
Comment (nickanderson) [inline api/enterprise-api-ref/two-factor-authentication.markdown:133]:
```suggestion
with the `Disable 2FA for any user` RBAC role (alias `2fa.any-user.disable`) will be able to
```

Make the alias verbatim

**Review by nickanderson — CHANGES_REQUESTED** [review]
Pretty good, mostly just agreeing with craigs responses.

File: api/enterprise-api-ref/two-factor-authentication.markdown (hunk 2 above)
Comment (aleksandrychev) [reply api/enterprise-api-ref/two-factor-authentication.markdown:42]: added output section

Comment (craigcomstock) [reply api/enterprise-api-ref/two-factor-authentication.markdown:42]: cool, thanks!

**Review by nickanderson — APPROVED** [review]

**Review by craigcomstock — APPROVED** [review]

**Comment by aleksandrychev** [comment]
@nickanderson please merge

**Pull request merged**